pennylane/plugins/default_gaussian.py

```diff
@@ -836,9 +836,43 @@ def var(self, observable, wires, par):
         _, var = self._observable_map[observable](mu, cov, wires, par, hbar=self.hbar, total_wires=self.num_wires)
         return var
 
+
     def sample(self, observable, wires, par, n=None):
-        raise NotImplementedError("Sampling is not supported in default.gaussian, "
-                                  "please install PennyLane-SF or another plugin capable of sampling")
+        r"""Returns n samples of the given observable in wires.
+
+        Args:
+            observable (string): Observable to be measured
+            wires (int of Sequence[int]): indices of the requested wires
+            par (list): Parameters for the observable
+            n (int): Number of samples to be generates
+
+        Returns:
+            array: samples
+        """
+        if n is None:
+            n = self.shots
+        if n <= 0 or not isinstance(n, int):
+            raise ValueError("The number of samples must be a positive integer.")
+
+        if len(wires) != 1:
+            raise ValueError("Only one mode can be measured in homodyne.")
+
+        if observable == "X":
+            phi = 0.0
+        elif observable == "P":
+            phi = np.pi/2
+        elif observable == "QuadOperator":
+            phi = par[0]
+        else:
+            raise NotImplementedError("Sampling for this observable is not implemented.")
```

Comment:
:+1:

Reply:
Might be worth also specifying the observable name in the error message:
```suggestion
            raise NotImplementedError("default.gaussian does not support sampling {}".format(observable))
```